core/services/pipeline/task.jsonparse.go

```diff
@@ -106,5 +109,21 @@ func (t *JSONParseTask) Run(_ context.Context, _ logger.Logger, vars Vars, input
 			return Result{Error: errors.Wrapf(ErrKeypathNotFound, `could not resolve path ["%v"] in %s`, strings.Join(path, `","`), data)}, runInfo
 		}
 	}
+
+	switch val := decoded.(type) {
+	case json.Number:
+		i64, err := val.Int64()
+		if err == nil {
+			decoded = i64
+		} else {
+			f64, err := val.Float64()
```

Comment:
Should/could we use a `decimal.Decimal` here instead? We could still return `float64` when exact: https://pkg.go.dev/github.com/shopspring/decimal#Decimal.Float64